File: 백준/Silver/11399. ATM/ATM.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bf.readLine());
        String[] strings = bf.readLine().split(" ");

        int[] arr = new int[n];
        for(int i=0; i<n; i++) {
            arr[i] = Integer.parseInt(strings[i]);
        }

        Arrays.sort(arr);

        int[] sum = new int[n];
        sum[0] = arr[0];
        for(int i=1; i<n; i++) {
            sum[i] = sum[i-1] + arr[i];
        }

        int ans = 0;
        for(int num : sum) {
            ans += num;
        }

        System.out.println(ans);

    }
}

File: 백준/Silver/11399. ATM/ATM.cc
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>

int Sort(int a[], int num);

int main()
{
	int n;
	int arr[1000] = {};
	scanf("%d", &n);
	for (int i = 0; i < n; i++)
	{
		scanf("%d", &arr[i]);
	}
	*arr = Sort(arr, n);
	int sum = 0;
	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j <= i; j++)
		{
			sum += arr[j];
		}
	}
	printf("%d\n", sum);
	return 0;
}

int Sort(int a[],int num)
{
	int *arr_sort = a;
	int temp;
	for (int i = 0; i < num; i++)
	{
		for (int j = 0; j < num-1; j++)
		{
			if (arr_sort[j] > arr_sort[j + 1])
			{
				temp = arr_sort[j];
				arr_sort[j] = arr_sort[j + 1];
				arr_sort[j + 1] = temp;
			}
		}
	}
	return *arr_sort;
}